src/persist/src/indexed/arrangement.rs

```diff
@@ -574,7 +574,7 @@ impl Arrangement {
         let since = self.since();
         let mut batches = Vec::with_capacity(self.trace_batches.len());
         for meta in self.trace_batches.iter() {
-            batches.push(blob.get_trace_batch_async(&meta.key));
+            batches.push(blob.get_trace_batch_async(&meta.key, true));
```

Comment:
what's the intuition behind snapshots saving to the cache?

Reply:
the thought process here was that on restarts, we'll be rereading blobs (because cache will be empty) and its likely worth it to cache those.

For unsealed - its definitely worth it because every unsealed batch will eventually have to get migrated into trace, and so saving any subset of unsealed batches into the cache prevents a cache miss sometime down the line.

For trace - its more unclear. Obviously, if the data is small enough to all fit in cache then writing it to cache is worth it to avoid cache misses on compaction. Otherwise, you could get lucky and fill the cache with good small trace batches that you'll be rereading for compaction soon, or you could get unlucky, and fill the cache with a large level 20 cache that blocks other trace batches from being cached for a long time.

I opted to have both unsealed and trace batches save to the cache on snapshot because it seemed like:

1. A simpler policy that we can tune down the line
2. I suspect that for realistic workloads / even for our tests the lower level trace batches will likely be larger than 2 GB and thus not fit in the trace cache anyway, so the downsides seemed not so bad.

Reply:
gotcha. I think the logic gets a touch more complicated if/when the cache becomes something on the filesystem and thus isn't empty on startup. OTOH, given how fast s3 is, I think we may be able to punt on that for quite a while

let's put your comment above as a comment in the code and call it a day

Reply:
done